clouddriver-aws/src/main/groovy/com/netflix/spinnaker/clouddriver/aws/provider/agent/AmazonClassicLoadBalancerCachingAgent.groovy

```diff
@@ -19,116 +19,32 @@ package com.netflix.spinnaker.clouddriver.aws.provider.agent
 
-class LoadBalancerCachingAgent implements CachingAgent, OnDemandAgent, AccountAware, DriftMetric {
-  final Logger log = LoggerFactory.getLogger(getClass())
-
-  private static final TypeReference<Map<String, Object>> ATTRIBUTES = new TypeReference<Map<String, Object>>() {}
-
-  private static final Collection<AgentDataType> types = Collections.unmodifiableCollection([
-    AUTHORITATIVE.forType(LOAD_BALANCERS.ns),
-    INFORMATIVE.forType(INSTANCES.ns)
-  ])
-
-  @Override
-  String getProviderName() {
-    AwsProvider.PROVIDER_NAME
-  }
-
-  @Override
-  String getAgentType() {
-    "${account.name}/${region}/${LoadBalancerCachingAgent.simpleName}"
-  }
-
-  @Override
-  String getAccountName() {
-    account.name
-  }
-
-  @Override
-  String getOnDemandAgentType() {
-    "${getAgentType()}-OnDemand"
-  }
-
-  @Override
-  Collection<AgentDataType> getProvidedDataTypes() {
-    types
-  }
-
-  final AmazonCloudProvider amazonCloudProvider
-  final AmazonClientProvider amazonClientProvider
-  final NetflixAmazonCredentials account
-  final String region
-  final ObjectMapper objectMapper
-  final Registry registry
-  final OnDemandMetricsSupport metricsSupport
-
-  LoadBalancerCachingAgent(AmazonCloudProvider amazonCloudProvider,
-                           AmazonClientProvider amazonClientProvider,
-                           NetflixAmazonCredentials account,
-                           String region,
-                           ObjectMapper objectMapper,
-                           Registry registry) {
-    this.amazonCloudProvider = amazonCloudProvider
-    this.amazonClientProvider = amazonClientProvider
-    this.account = account
-    this.region = region
-    this.objectMapper = objectMapper.enable(SerializationFeature.WRITE_DATES_AS_TIMESTAMPS)
-    this.registry = registry
-    this.metricsSupport = new OnDemandMetricsSupport(registry, this, amazonCloudProvider.id + ":" + "${amazonCloudProvider.id}:${OnDemandAgent.OnDemandType.LoadBalancer}")
-  }
-
-  static class MutableCacheData implements CacheData {
-    final String id
-    int ttlSeconds = -1
-    final Map<String, Object> attributes = [:]
-    final Map<String, Collection<String>> relationships = [:].withDefault { [] as Set }
-    public MutableCacheData(String id) {
-      this.id = id
-    }
-
-    @JsonCreator
-    public MutableCacheData(@JsonProperty("id") String id,
-                            @JsonProperty("attributes") Map<String, Object> attributes,
-                            @JsonProperty("relationships") Map<String, Collection<String>> relationships) {
-      this(id);
-      this.attributes.putAll(attributes);
-      this.relationships.putAll(relationships);
-    }
-  }
-
-  @Override
-  boolean handles(OnDemandAgent.OnDemandType type, String cloudProvider) {
-    type == OnDemandAgent.OnDemandType.LoadBalancer && cloudProvider == amazonCloudProvider.id
+class AmazonClassicLoadBalancerCachingAgent extends AbstractAmazonLoadBalancerCachingAgent {
```

Comment:
I actually think it would be worth not changing the classname - (we've already got some other less that spectacularly named caching agents in clouddriver-aws) - just to keep the roll in/roll back story simpler